sys/arm64/include/cherireg.h

```diff
@@ -0,0 +1,158 @@
+/*-
+ * Copyright (c) 2011-2017 Robert N. M. Watson
+ * Copyright (c) 2016-2020 Andrew Turner
+ * All rights reserved.
+ *
+ * This software was developed by SRI International and the University of
+ * Cambridge Computer Laboratory under DARPA/AFRL contract (FA8750-10-C-0237)
+ * ("CTSRD"), as part of the DARPA CRASH research programme.
+ *
+ * This work was supported by Innovate UK project 105694, "Digital Security
+ * by Design (DSbD) Technology Platform Prototype".
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * $FreeBSD$
+ */
+#ifndef _ARM64_INCLUDE_CHERIREG_H_
+#define	_ARM64_INCLUDE_CHERIREG_H_
+
+#define	CHERICAP_SIZE   16
+#define	CHERICAP_SHIFT	4
+
+/*
+ * CHERI ISA-defined constants for capabilities -- suitable for inclusion from
+ * assembly source code.
+ */
+#define	CHERI_PERM_GLOBAL			(1 << 0)	/* 0x00000001 */
```

Comment:
Can we just use the compiler-provided macros instead of duplicating the values here?

Reply:
This matches what we do on MIPS and RISC-V.  However, we should perhaps define permissions using the compiler constant in the MI sys/cheri/cherireg.h and only bother with truly MD permissions (Morello has a few I think) in the MD cherireg.h.  I don't think that needs to be part of this PR though, but can be done as a later cleanup across all 3 architectures.

Reply:
We should probably start including cheriintrin.h since that defines all portable permissions.

Reply:
That is a much harder change due to the use of `-nostdinc` for the kernel by default.  Removing that will likely expose various issues with clang's built in headers vs the kernel's.